script_mteb_french/results_analysis/compute_avergae_performance.py

```diff
@@ -0,0 +1,73 @@
+import os
+import sys
+import argparse
+import logging
+
+from datasets import load_dataset
+import pandas as pd
+
+from results_parser import ResultsParser
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+
+def split_model_name(model_name: str):
+    model_name = model_name.split('/')
+    if len(model_name) == 2:
+        model_name = model_name[1]
+    else:
+        model_name = model_name[0]
+    return model_name
+
+
+def main(args):
+    res_parser = ResultsParser()
+    fr_results = res_parser(args.results_folder)
+    
+    task_types = [task_type for task_type, _ in set(fr_results.columns.values)]
+
+    models_name_to_index = {idx: split_model_name(idx) for idx in fr_results.index}
+
+    fr_results_avg = fr_results.copy()
+    new_df = pd.DataFrame({})
+
+    for task_type in set(task_types):
+        filtered_results = fr_results_avg.loc[:][task_type]
+        new_df[f"avg_{task_type}"] = filtered_results.mean(axis=1)
+    
+    overall_avg = new_df.mean(axis=1)
+    new_df.insert(0, "overall_avg", overall_avg)
+
+    new_df.reset_index(inplace=True)
+    models_short_name = new_df.model.apply(lambda x: models_name_to_index[x])
```

Comment:
Why storing the index mapping, then computing on a new dataframe, then reseting the index ? You can compute directly averages on level 0, and change the columns name by prefixing with "avg_"
```py
# Compute average on each task type
averaged_df = result_df.T.groupby(level=0).mean().T
# change column names by prefixing with "avg_"
averaged_df.columns = [f"avg_{col.lower()}" for col in averaged_df.columns]
# compute overall average
averaged_df.insert(0, "avg_overall", averaged_df.mean(axis=1))
# only keep model name in path
averaged_df.index = averaged_df.index.map(os.path.basename)
```